src/client/src/commons/form/borehole/hydrogeology/waterIngressInput.js

```diff
@@ -0,0 +1,329 @@
+import React, { useEffect, useContext, forwardRef } from "react";
+import { useForm, Controller } from "react-hook-form";
+import {
+  Box,
+  Card,
+  Checkbox,
+  FormControlLabel,
+  FormControl,
+  InputLabel,
+  Input,
+  MenuItem,
+  Select,
+  Stack,
+  TextField,
+  Tooltip,
+} from "@mui/material";
+import { useDomains } from "../../../../api/fetchApiV2";
+import ClearIcon from "@mui/icons-material/Clear";
+import { useTranslation } from "react-i18next";
+import { AlertContext } from "../../../alert/alertContext";
+import { styled } from "@mui/system";
+
+const WaterIngressInput = props => {
+  const {
+    waterIngress,
+    setSelectedWaterIngress,
+    boreholeId,
+    addWaterIngress,
+    updateWaterIngress,
+  } = props;
+  const domains = useDomains();
+  const { t, i18n } = useTranslation();
+  const {
+    register,
+    handleSubmit,
+    control,
+    getValues,
+    formState: { errors },
+    trigger,
+  } = useForm();
+  const alertContext = useContext(AlertContext);
+
+  // submit values on unmount with useEffect clean up function.
+  useEffect(() => {
+    return () => {
+      handleSubmit(handleFormSubmit)();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [handleSubmit]);
+
+  // trigger form validation on mount
+  useEffect(() => {
+    trigger();
+  }, [trigger]);
+
+  const handleFormSubmit = data => {
+    //convert dates to IsoStrings
+    data?.startTime ? (data.startTime += ":00.000Z") : (data.startTime = null);
+    data?.endTime ? (data.endTime += ":00.000Z") : (data.endTime = null);
+    if (data.startTime && data.quantityId && data.reliabilityId) {
+      if (waterIngress.id === 0) {
+        addWaterIngress({ ...data, type: 1, boreholeId: boreholeId });
```

Comment:
Q: Handelt es sich bei `type` um den `ObservationType`? Fall ja, bitte den Wert in eine Variable/Konstante schieben und dann übergeben, damit man auf den ersten Blick sieht, um was für einen Typen es sich handelt.

Reply:
ja! Sobald es mehrere gibt kann ich dann auch ein enum machen.